File: plugins/gson/plugin/src/main/java/com/mckj/annotation/utlis/HexEncryptUtils.java
package com.mckj.annotation.utlis;

public class HexEncryptUtils {

    public static byte[] encrypt(byte[] bytes, int keys) {
        if (bytes == null) {
            return null;
        }
        int len = bytes.length;
        int key = (keys * keys * keys) << 1;
        int pos;
        for (int i = 0; i < len; i++) {
            if (i % 2 == 0) {
                pos = (i * i) >> 1;
            } else {
                pos = (i * i * i) >> 1;
            }
            key = key ^ (pos);
            bytes[i] = (byte) (bytes[i] ^ key);
            key = bytes[i];
        }
        return bytes;
    }
}

File: plugins/stringEncrypt/plugin/src/main/kotlin/com/plugin/stringEncrypt/encrypt/HexEncryptUtils.kt
package com.plugin.stringEncrypt.encrypt

import kotlin.experimental.xor

/**
 * HexEncryptUtils
 *
 * @author mmxm
 * @date 2023/7/20 17:50
 */
object HexEncryptUtils {

    fun encrypt(bytes: ByteArray?, keys: Int): ByteArray? {
        if (bytes == null) {
            return null
        }
        val len = bytes.size
        var key = keys * keys * keys shl 1
        var pos: Int
        for (i in 0 until len) {
            pos = if (i % 2 == 0) {
                i * i shr 1
            } else {
                i * i * i shr 1
            }
            key = key xor pos
            bytes[i] = (bytes[i].toInt() xor key).toByte()
            key = bytes[i].toInt()
        }
        return bytes
    }
}